core/store/migrate/migrate_test.go

```diff
@@ -0,0 +1,265 @@
+		Type                           job.Type
+		SchemaVersion                  uint32
+		Name                           null.String
+		MaxTaskDuration                models.Interval
+		Pipeline                       pipeline.Pipeline
+		CreatedAt                      time.Time
+	}
+
+	spec := OffchainReporting2OracleSpec{
+		ID:                                100,
+		ContractID:                        "terra_187246hr3781h9fd198fh391g8f924",
+		Relay:                             "terra",
+		RelayConfig:                       job.RelayConfig{},
+		P2PBootstrapPeers:                 pq.StringArray{""},
+		OCRKeyBundleID:                    null.String{},
+		MonitoringEndpoint:                null.StringFrom("endpoint:chainlink.monitor"),
+		TransmitterID:                     null.String{},
+		BlockchainTimeout:                 1337,
+		ContractConfigTrackerPollInterval: 16,
+		ContractConfigConfirmations:       32,
+		JuelsPerFeeCoinPipeline:           "",
+		IsBootstrapPeer:                   true,
+	}
+	spec2 := OffchainReporting2OracleSpec{
+		ID:                                200,
+		ContractID:                        "sol_187246hr3781h9fd198fh391g8f924",
+		Relay:                             "sol",
+		RelayConfig:                       job.RelayConfig{},
+		P2PBootstrapPeers:                 pq.StringArray{""},
+		OCRKeyBundleID:                    null.String{},
+		MonitoringEndpoint:                null.StringFrom("endpoint:chain.link.monitor"),
+		TransmitterID:                     null.String{},
+		BlockchainTimeout:                 1338,
+		ContractConfigTrackerPollInterval: 17,
+		ContractConfigConfirmations:       33,
+		JuelsPerFeeCoinPipeline:           "",
+		IsBootstrapPeer:                   true,
+	}
+
+	jb := Job{
+		ID:                             10,
+		ExternalJobID:                  uuid.NewV4(),
+		Type:                           job.OffchainReporting2,
+		SchemaVersion:                  1,
+		PipelineSpecID:                 pipelineID,
+		Offchainreporting2OracleSpecID: &spec.ID,
+		Offchainreporting2OracleSpec:   &spec,
+	}
+
+	jb2 := Job{
+		ID:                             20,
+		ExternalJobID:                  uuid.NewV4(),
+		Type:                           job.OffchainReporting2,
+		SchemaVersion:                  1,
+		PipelineSpecID:                 pipelineID2,
+		Offchainreporting2OracleSpecID: &spec2.ID,
+		Offchainreporting2OracleSpec:   &spec2,
+	}
+
+	nonBootstrapSpec := OffchainReporting2OracleSpec{
+		ID:                101,
+		IsBootstrapPeer:   false,
+		P2PBootstrapPeers: pq.StringArray{""},
+		ContractID:        "empty",
+	}
+	nonBootstrapJob := Job{
+		ID:                             11,
+		ExternalJobID:                  uuid.NewV4(),
+		Type:                           job.OffchainReporting2,
+		SchemaVersion:                  1,
+		PipelineSpecID:                 nonBootstrapPipelineID,
+		Offchainreporting2OracleSpecID: &nonBootstrapSpec.ID,
+		Offchainreporting2OracleSpec:   &nonBootstrapSpec,
+	}
```

Comment:
maybe add a 4th job which is already in the new format to ensure no bad collisions/overwrites?